Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -1488,17 +1490,8 @@ - (BOOL)isCurrentArticleSaved {
 #pragma mark Scroll to last section after rotate
 
 - (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration {
-    NSString* js = @"(function() {"
-                   @"    _topElement = document.elementFromPoint( window.innerWidth / 2, 0 );"
-                   @"    if (_topElement) {"
-                   @"        var rect = _topElement.getBoundingClientRect();"
-                   @"        return rect.top / rect.height;"
-                   @"    } else {"
-                   @"        return 0;"
-                   @"    }"
-                   @"})()";
-    float relativeScrollOffset = [[self.webView stringByEvaluatingJavaScriptFromString:js] floatValue];
-    self.relativeScrollOffsetBeforeRotate = relativeScrollOffset;
+    [[self.webView wmf_javascriptContext][@"setPreRotationRelativeScrollOffset"] callWithArguments:nil];
```

Comment:
No it's what determines the offset. Turned out the js "_topElement" var was keeping state, so since it was already doing this I just made it also keep "_preRotationOffsetY" so we didn't have to do so in Obj-C. Not ideal...